tests/dns/static_dns/conftest.py

```diff
@@ -0,0 +1,109 @@
+import pytest
+import logging
+
+from tests.common.plugins.allure_wrapper import allure_step_wrapper as allure
+from .static_dns_util import RESOLV_CONF_FILE, get_nameserver_from_config_db, get_nameserver_from_resolvconf, \
+    config_mgmt_ip, clear_nameserver_from_resolvconf
+
+logger = logging.getLogger(__name__)
+allure.logger = logger
+
+
+def pytest_addoption(parser):
+    """
+        Adds options to pytest that are used by the rif loopback action tests.
+    """
+
+    parser.addoption(
+        "--static_dns_reboot_type",
+        action="store",
+        type=str,
+        default="cold",
+        help="reboot type such as reload, cold, fast, warm, random"
+    )
+
+    parser.addoption(
+        "--both_static_dynamic_ip_supported",
+        action="store",
+        type=bool,
+        default=False,
+        help="Both dynamic and static ip are configured on the dut"
+    )
+
+
+@pytest.fixture(scope="module", autouse=True)
+def is_static_dns_supported(duthost):
+    cmd_err = duthost.shell("show dns nameserver", module_ignore_errors=True)['stderr']
+    if 'Error: No such command "dns"' in cmd_err:
+        pytest.skip("The static DNS is not supported by this image.")
+
+
+@pytest.fixture(scope="module", autouse=True)
+def static_dns_setup(duthost):
+    with allure.step("Get all existing DNS nameserver from config db"):
+        nameservers_db = get_nameserver_from_config_db(duthost)
+
+        duthost.shell(f"cp {RESOLV_CONF_FILE} {RESOLV_CONF_FILE}.bk")
```

Comment:
Did you recover resolv.conf after this test?

Reply:
Hi @ganglyu 

We do not need to recover the resolv.conf file, after the dns nameservers added back to the config db, the resolv.conf file could be updated automatically, the back up of the resolv.conf file is only used in "test_static_dns_basic", it is used to verify after the nameservers added back to the config db, the resolv.conf file will be updated to have same content as the backup file.

Reply:
We could have other tests that will use resolv.conf and name server in CONFIG_DB, please make sure that they are recovered after the test.

Reply:
yes, they are recovered when we execute :
            yield
            
            with allure.step("Recover DNS nameserver in config db"):
                   for nameserver in nameservers:
                       duthost.shell(f"config dns nameserver add {nameserver}")